core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetAggregationUtils.java

```diff
@@ -97,9 +61,9 @@ public static void combineState(@AggregationState HyperLogLogState state, @Aggre
         }
     }
 
-    @OutputFunction(StandardTypes.HYPER_LOG_LOG)
     public static void evaluateFinal(@AggregationState HyperLogLogState state, BlockBuilder out)
     {
+        getOrCreateHyperLogLog(state); // ensure we have some state even if input was empty
```

Comment:
why is it needed now?

Reply:
I will drop it for now. It actually is a semantic change. The current semantic is that `approx_set` will return NULL if input is empty or contains just NULLs. It is weird, but maybe it is how it supposed to be.
@martint ?

Reply:
> The current semantic is that `approx_set` will return NULL if input is empty or contains just NULLs. It is weird, but maybe it is how it supposed to be.

Let's have a separate issue for this.
(Do we have a test?)

Reply:
Yeah - it is covered. I will create issue if @martint confirms that current behaviour is broken. IDK to be honest.

Reply:
IDK either. We can have an issue still and then close as 'not a bug'.

Reply:
https://github.com/trinodb/trino/issues/8347